chips/nrf51/src/uart.rs

```diff
@@ -142,7 +159,25 @@ impl UART {
             }
         }
         if tx {
-            // Should never execute!
+            regs.event_txdrdy.set(0 as u32);
+
+            if self.len.get() == self.index.get() {
+                regs.task_stoptx.set(1 as u32);
+
+                // Signal client write done
+                match self.client {
+                    Some(ref client) => client.write_done(self.buffer.take().unwrap()),
```

Comment:
No `unwrap`. Use `map` or `unwrap_or` instead. `unwrap` can panic and we should avoid potential panics whenever possible.